vulnfeeds/cmd/cvelist2osv/main.go

```diff
@@ -0,0 +1,293 @@
+package main
+
+import (
+	"encoding/json"
+	"errors"
+	"flag"
+	"fmt"
+	"os"
+	"path/filepath"
+	"strings"
+	"time"
+
+	"golang.org/x/exp/slices"
+
+	"github.com/google/osv/vulnfeeds/cves"
+	"github.com/google/osv/vulnfeeds/git"
+	"github.com/google/osv/vulnfeeds/utility"
+	"github.com/google/osv/vulnfeeds/vulns"
+	"github.com/ossf/osv-schema/bindings/go/osvschema"
+)
+
+type ConversionOutcome int
+
+var ErrNoRanges = errors.New("no ranges")
+
+var ErrUnresolvedFix = errors.New("fixes not resolved to commits")
+
+// String returns the string representation of a ConversionOutcome.
+func (c ConversionOutcome) String() string {
+	return [...]string{
+		"ConversionUnknown",
+		"Successful",
+		"Rejected",
+		"NoSoftware",
+		"NoRepos",
+		"NoRanges",
+		"FixUnresolvable"}[c]
+}
+
+const (
+	extension = ".json"
+)
+
+const (
+	// Set of enums for categorizing conversion outcomes.
+	ConversionUnknown ConversionOutcome = iota // Shouldn't happen
+	Successful                                 // It worked!
+	Rejected                                   // The CVE was rejected
+	NoSoftware                                 // The CVE had no CPEs relating to software (i.e. Operating Systems or Hardware).
+	NoRepos                                    // The CPE Vendor/Product had no repositories derived for it.
+	NoRanges                                   // No viable commit ranges could be calculated from the repository for the CVE's CPE(s).
+	FixUnresolvable                            // Partial resolution of versions, resulting in a false positive.
+)
+
+var (
+	jsonPath  = flag.String("cve_json", "", "Path to CVEList JSON to examine.")
+	outDir    = flag.String("out_dir", "", "Path to output results.")
+	outFormat = flag.String("out_format", "OSV", "Format to output {OSV}")
+)
+var Logger utility.LoggerWrapper
+var RepoTagsCache git.RepoTagsCache
+var Metrics struct {
+	CNA           string
+	Outcome       ConversionOutcome
+	RefTypesCount map[osvschema.ReferenceType]int
+	Notes         []string
+}
+
+// RefTagDenyList: References with these tags have been found to contain completely unrelated
+// repositories and can be misleading as to the software's true repository,
+// Currently not used for this purpose due to undesired false positives
+// reducing the number of valid records successfully converted.
+var RefTagDenyList = []string{
+	// "Exploit",
+	// "Third Party Advisory",
+	"Broken Link", // Actively ignore these though.
+}
+
+// addIDToNotes prepends a CVE ID to a slice of notes.
+func addIDToNotes(id cves.CVEID, notes []string) []string {
+	var updatedNotes []string
+	for _, note := range notes {
+		updatedNotes = append(updatedNotes, fmt.Sprintf("[%s] %s", id, note))
+	}
+	return updatedNotes
+}
+
+// determineHeuristics examines a CVE and extracts metrics and heuristics about it.
```

Comment:
Rename determineHeuristics to something more specifically about extracting metrics, and clarify that this does not modify the vulnerability. (I would even avoid taking the vulnerability by pointer, but that might have performance implications)